src/Dapr.Jobs/Models/Responses/JobDetails.cs

```diff
@@ -0,0 +1,89 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+
+namespace Dapr.Jobs.Models.Responses;
+
+/// <summary>
+/// Represents the details of a retrieved job.
+/// </summary>
+public sealed record JobDetails
+{
+    /// <summary>
+    /// Regular expression used to determine if a given schedule is a Cron expression or an interval.
+    /// </summary>
+    private readonly Regex isIntervalRegex = new("h|m|(ms)|s");
+
+    ///<summary>
+    /// A cron-like expression that defines when a job should be triggered.
+    /// </summary>
+    /// <remarks>
+    /// Either this or the <see cref="Interval"/> property should be specified.
+    /// </remarks>
+    public string? CronExpression => Schedule is null || !isIntervalRegex.IsMatch(Schedule) ? Schedule : null;
+
+    /// <summary>
+    /// The interval expression that defines when a job should be triggered.
+    /// </summary>
+    /// <remarks>
+    /// Either this or the <see cref="CronExpression"/> property should be specified.
+    /// </remarks>
+    public TimeSpan? Interval =>
+        Schedule is not null && isIntervalRegex.IsMatch(Schedule) ? Schedule.FromDurationString() : null;
+
+    /// <summary>
+    /// Represents whether the job is scheduled using a Cron expression.
+    /// </summary>
+    public bool IsCronExpression => CronExpression is not null;
+
+    /// <summary>
+    /// Represents whether the job is scheduled using an interval expression.
+    /// </summary>
+    public bool IsIntervalExpression => !IsCronExpression;
+
+    /// <summary>
+    /// The string-based schedule value returned by the job details payload.
+    /// </summary>
+    [JsonPropertyName("schedule")]
+    public string? Schedule { get; init; } = null;
+
+    /// <summary>
+    /// Allows for jobs with fixed repeat counts.
+    /// </summary>
+    [JsonPropertyName("repeats")]
+    public uint? RepeatCount { get; init; } = null;
```

Comment:
Nit: the guidelines says APIs should use `int` unless there's an explicit reason not to (such as you need the extra value space). This is despite it being the more technically correct type.  A long-standing rule, I believe, for the benefit of non-C# .NET languages that do not necessarily have the same signed vs. unsigned support.

Reply:
That's fair - updated. Habit from my own development practice.

Reply:
I was refactoring another part of this when I realized specifically where I'd gotten this from here beyond my own practice. The generated prototype client uses uint requiring [this cast](https://github.com/dapr/dotnet-sdk/blob/13c32ded3f0acba6df73f414ad1e68698fe04596/src/Dapr.Jobs/DaprJobsGrpcClient.cs#L88)

Regardless, I think this is worth exposing in this SDK as int and just having that range checker.